### PR TITLE
Solves root is false is not getting followed

### LINKED ISSUE #26130: roots = False not being followed
from sympy import *
import sympy
from sympy.abc import k, m, n, x
print(roots(-64*x**7 + 112*x**5 - 56*x**3 + 7*x, trig = False))

Running this code gives a trigonometric expression, even if (trig = False) is called

Code output snippet: 
{-sqrt(6)*(-56*sqrt(7)*cos(atan(3*sqrt(3))/3)**3 - 56*sqrt(7)*cos(atan(3*sqrt(3))/3)**3*cos(2*atan(3*sqrt(3))/3) - 56*sqrt(7)*sin(atan(3*sqrt(3))/3)*sin(2*atan(3*sqrt(3))/3)*cos(atan(3*sqrt(3))/3)**2 - 56*sqrt(7)*sin(atan(3*sqrt(3))/3)**2*cos(atan(3*sqrt(3))/3) - 56*sqrt(7)*sin(atan(3*sqrt(3))/3)**2*cos(atan(3*sqrt(3))/3)*cos(2*atan(3*sqrt(3))/3) - 56*sqrt(7)*sin(atan(3*sqrt(3))/3)**3*sin(2*atan(3*sqrt(3))/3) + 28*sin(atan(3*sqrt(3))/3)**2*cos(2*atan(3*sqrt(3))/3)**2 + 28*sin(atan(3*sqrt(3))/3)**2*sin(2*atan(3*sqrt(3))/3)**2 + 28*sin(atan(3*sqrt(3))/3)**2 + 56*sin(atan(3*sqrt(3))/3)**2*cos(2*atan(3*sqrt(3))/3) +...

### PR DESCRIPTION
**Pull request by abhiphile**
#### References to other Issues or PRs
Fixes #26130 

#### Brief description of what is fixed or changed
Solved the issue of not giving expression without trigonometric identities even after passing ` trig = False `

#### Release Notes

* polys
  * Fixed a bug in roots function in which ` trig = False ` is not getting followed.

### CODE REVIEW

File: sympy/polys/polyroots.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def roots_linear(f):
     return [r]
 
 
-def roots_quadratic(f):
+def roots_quadratic(f,trig = False):
```
Comment (sylee957) [inline sympy/polys/polyroots.py:52]:
```suggestion
def roots_quadratic(f, trig=False):
```

File: sympy/polys/polyroots.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def _roots_quartic_euler(p, q, r, a):
     return [c1 - c2 - a, -c1 - c3 - a, -c1 + c3 - a, c1 + c2 - a]
 
 
-def roots_quartic(f):
+def roots_quartic(f,trig = False):
```
Comment (sylee957) [inline sympy/polys/polyroots.py:247]:
```suggestion
def roots_quartic(f, trig=False):
```

File: sympy/polys/polyroots.py [hunk 3]
```diff
@@ -368,7 +368,7 @@ def _ans(y):
                     for a1, a2 in zip(_ans(y1), _ans(y2))]
 
 
-def roots_binomial(f):
+def roots_binomial(f,trig=False):
```
Comment (sylee957) [inline sympy/polys/polyroots.py:371]:
```suggestion
def roots_binomial(f, trig=False):
```

File: sympy/polys/polyroots.py [hunk 4]
```diff
@@ -381,7 +381,7 @@ def roots_binomial(f):
     base = -cancel(b/a)
     alpha = root(base, n)
 
-    if alpha.is_number:
+    if alpha.is_number and trig == True:
```
Comment (sylee957) [inline sympy/polys/polyroots.py:384]:
```suggestion
    if alpha.is_number and trig:
```

File: sympy/polys/polyroots.py [hunk 5]
```diff
@@ -1029,7 +1029,7 @@ def _try_heuristics(f):
             if f.degree() == 1:
                 return list(map(cancel, roots_linear(f)))
             else:
-                return roots_binomial(f)
+                return roots_binomial(f,trig = trig)
```
Comment (sylee957) [inline sympy/polys/polyroots.py:1032]:
```suggestion
                return roots_binomial(f, trig=trig)
```

File: sympy/polys/polyroots.py [hunk 6]
```diff
@@ -1090,7 +1090,7 @@ def _try_heuristics(f):
             _update_dict(result, zeros, roots_linear(f)[0], 1)
         elif f.length() == 2:
             roots_fun = roots_quadratic if f.degree() == 2 else roots_binomial
-            for r in roots_fun(f):
+            for r in roots_fun(f,trig = trig):
```
Comment (sylee957) [inline sympy/polys/polyroots.py:1093]:
```suggestion
            for r in roots_fun(f, trig=trig):
```

File: sympy/polys/tests/test_polyroots.py [hunk 7]
```diff
@@ -756,3 +756,15 @@ def test_issue_22768():
         r: 1,
         -r*(1 + sqrt(3)*I)/2: 1,
         r*(-1 + sqrt(3)*I)/2: 1}
+
+def test_issue_26130():
+    assert roots(-64*x**7 + 112*x**5 - 56*x**3, trig = False) == {
```
Comment (sylee957) [inline sympy/polys/tests/test_polyroots.py:761]:
```suggestion
    assert roots(-64*x**7 + 112*x**5 - 56*x**3, trig=False) == {
```

File: sympy/polys/tests/test_polyroots.py [hunk 8]
```diff
@@ -756,3 +756,15 @@ def test_issue_22768():
...
+        -sqrt(7/8 - sqrt(7)*I/8): 1,
+        sqrt(7/8 - sqrt(7)*I/8): 1,
+        -sqrt(7/8 + sqrt(7)*I/8): 1,
+        sqrt(7/8 + sqrt(7)*I/8): 1, 0: 3}
+    assert (roots(-64*x**7 + 112*x**5 - 56*x**3, trig = True)) == {
```
Comment (sylee957) [inline sympy/polys/tests/test_polyroots.py:766]:
```suggestion
    assert roots(-64*x**7 + 112*x**5 - 56*x**3, trig=True) == {
```

File: sympy/polys/polyroots.py [hunk 9]
```diff
@@ -381,7 +381,7 @@ def roots_binomial(f):
     base = -cancel(b/a)
     alpha = root(base, n)
 
-    if alpha.is_number:
+    if alpha.is_number and trig:
         alpha = alpha.expand(complex=True)
```
Comment (oscarbenjamin) [inline sympy/polys/polyroots.py:385]: It would be better to remove the call to `expand` altogether I think.

Comment (abhiphile) [reply sympy/polys/polyroots.py:385]: I've tried doing it but some of the test cases failed 
``` FAILED sympy/polys/tests/test_polyroots.py::test_issue_8285 - assert False
FAILED sympy/polys/tests/test_polyroots.py::test_issue_8289 - assert False
FAILED sympy/polys/tests/test_polyroots.py::test_roots_quartic - assert [(-1)**(1/3),.../6)*sqrt(3)/2] == [-1, 0, 1/2 +...- sqrt(3)*I/2]
FAILED sympy/polys/tests/test_polyroots.py::test_roots_binomial - assert [-(-2)**(1/4)...4)*5**(3/4)/5] == [-10**(3/4)/1...0**(3/4)*I/10]
FAILED sympy/polys/tests/test_polyroots.py::test_issue_20913 - assert [-(-1)**(1/3).../3)*sqrt(3)/2] == [-2**(2/3)] 
```

Comment (abhiphile) [reply sympy/polys/polyroots.py:385]: I've tried doing it but some of the test cases failed 
``` FAILED sympy/polys/tests/test_polyroots.py::test_issue_8285 - assert False
FAILED sympy/polys/tests/test_polyroots.py::test_issue_8289 - assert False
FAILED sympy/polys/tests/test_polyroots.py::test_roots_quartic - assert [(-1)**(1/3),.../6)*sqrt(3)/2] == [-1, 0, 1/2 +...- sqrt(3)*I/2]
FAILED sympy/polys/tests/test_polyroots.py::test_roots_binomial - assert [-(-2)**(1/4)...4)*5**(3/4)/5] == [-10**(3/4)/1...0**(3/4)*I/10]
FAILED sympy/polys/tests/test_polyroots.py::test_issue_20913 - assert [-(-1)**(1/3).../3)*sqrt(3)/2] == [-2**(2/3)] ```

Comment (oscarbenjamin) [reply sympy/polys/polyroots.py:385]: I have a more complete PR that changes this although it also changes other things gh-25196.

**Pull request closed without merging**